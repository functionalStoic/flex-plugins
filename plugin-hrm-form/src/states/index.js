import { combineReducers } from 'redux';

import { reduce as ContactStateReducer } from './ContactState';

// Register your redux store under a unique namespace
export const namespace = 'plugin-hrm-form';
export const contactFormsBase = 'activeContacts';

// Combine the reducers
export default combineReducers({
  [contactFormsBase]: ContactStateReducer,
});
