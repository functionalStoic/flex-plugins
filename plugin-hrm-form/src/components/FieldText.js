import React from 'react';
import { StyledInput,
  StyledLabel,
  ErrorText,
  TextField
} from '../Styles/HrmStyles';
import { ValidationType } from '../states/ContactFormStateFactory';

const RequiredAsterisk = ({ field }) => {
  const isRequired = field.validation && field.validation.includes(ValidationType.REQUIRED);
  return isRequired && <span style={{color: 'red'}}>*</span>;
}

const FieldText = ({ label, parents, name, field, taskId, handleBlur, handleChange, handleFocus }) => {
  const id = [...parents, name].join('_');

  return (
    <TextField>
      <StyledLabel htmlFor={id}>
        {label}<RequiredAsterisk field={field} />
      </StyledLabel>
      <StyledInput
        error={field.error !== null}
        name={name}
        id={id}
        value={field.value}
        onBlur={handleBlur}
        onChange={(e) => handleChange(taskId, parents, e)}
        onFocus={() => handleFocus(taskId, parents, name)}
      />
      {field.error && <ErrorText>{field.error}</ErrorText>}
    </TextField>
  );
};
          
export default FieldText;
