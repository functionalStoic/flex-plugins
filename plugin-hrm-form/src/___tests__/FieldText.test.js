jest.mock('../Styles/HrmStyles', () => {
    return {
      StyledInput: 'StyledInput',
      StyledLabel: 'StyledLabel',
      ErrorText: 'ErrorText',
      TextField: 'TextField'
    };
  });
  import React from 'react';
  import FieldText from '../components/FieldText';
  import { ValidationType } from '../states/ContactFormStateFactory';
  import renderer from 'react-test-renderer';
  
  //TODO: Improve assertion for all tests below
  test('render basic FieldText', () => {
    const firstName = {
      value: '',
      touched: false,
      error: null,
      validation: null
    };
    const handleBlur = jest.fn();
    const handleChange = jest.fn();
    const handleFocus = jest.fn();
    const taskId = '';

    const component = renderer.create(
      <FieldText
        label="First name"
        parents={['callerInformation', 'name']}
        name="firstName"
        field={firstName}
        taskId={taskId}
        handleBlur={handleBlur}
        handleChange={handleChange}
        handleFocus={handleFocus}
      />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  
  test('render required FieldText', () => {
    const firstName = {
      value: '',
      touched: false,
      error: null,
      validation: [ ValidationType.REQUIRED ]
    };
    const handleBlur = jest.fn();
    const handleChange = jest.fn();
    const handleFocus = jest.fn();
    const taskId = '';

    const component = renderer.create(
      <FieldText
        label="First name"
        parents={['callerInformation', 'name']}
        name="firstName"
        field={firstName}
        taskId={taskId}
        handleBlur={handleBlur}
        handleChange={handleChange}
        handleFocus={handleFocus}
      />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  
  
  test('render FieldText with errors', () => {
    const firstName = {
      value: '',
      touched: true,
      error: 'This field is required',
      validation: [ ValidationType.REQUIRED ]
    };
    const handleBlur = jest.fn();
    const handleChange = jest.fn();
    const handleFocus = jest.fn();
    const taskId = '';

    const component = renderer.create(
      <FieldText
        label="First name"
        parents={['callerInformation', 'name']}
        name="firstName"
        field={firstName}
        taskId={taskId}
        handleBlur={handleBlur}
        handleChange={handleChange}
        handleFocus={handleFocus}
      />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  