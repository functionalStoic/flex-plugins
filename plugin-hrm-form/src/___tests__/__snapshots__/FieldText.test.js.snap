// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render FieldText with errors 1`] = `
<TextField>
  <StyledLabel
    htmlFor="callerInformation_name_firstName"
  >
    First name
    <span
      style={
        Object {
          "color": "red",
        }
      }
    >
      *
    </span>
  </StyledLabel>
  <StyledInput
    error={true}
    id="callerInformation_name_firstName"
    name="firstName"
    onBlur={[MockFunction]}
    onChange={[Function]}
    onFocus={[Function]}
    value=""
  />
  <ErrorText>
    This field is required
  </ErrorText>
</TextField>
`;

exports[`render basic FieldText 1`] = `
<TextField>
  <StyledLabel
    htmlFor="callerInformation_name_firstName"
  >
    First name
  </StyledLabel>
  <StyledInput
    error={false}
    id="callerInformation_name_firstName"
    name="firstName"
    onBlur={[MockFunction]}
    onChange={[Function]}
    onFocus={[Function]}
    value=""
  />
</TextField>
`;

exports[`render required FieldText 1`] = `
<TextField>
  <StyledLabel
    htmlFor="callerInformation_name_firstName"
  >
    First name
    <span
      style={
        Object {
          "color": "red",
        }
      }
    >
      *
    </span>
  </StyledLabel>
  <StyledInput
    error={false}
    id="callerInformation_name_firstName"
    name="firstName"
    onBlur={[MockFunction]}
    onChange={[Function]}
    onFocus={[Function]}
    value=""
  />
</TextField>
`;
